torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -239,17 +287,127 @@ void ComputeAtMap::mapIds(IterDomain* id0, IterDomain* id1) {
   }
 }
 
-void ComputeAtMap::build(Fusion* fusion, GpuLower* gpu_lower) {
+void ComputeAtMap::build(
+    Fusion* fusion,
+    const TrivialReductionInfo* trivial_reduction_info) {
+  TORCH_INTERNAL_ASSERT(
+      trivial_reduction_info != nullptr,
+      "Trivial reduction info needs to be constructed and passed in to compute at map build.");
+
   // Consumers can only show up once in an expression, keep track of all of them
   std::vector<TensorView*> consumer_tvs;
 
+  // Track all domains that are the result of a view operation and consumed in a
+  // subsequent expression. When resolving concrete IDs we want to set these
+  // domains to count of concrete domains = 1 so they won't be favored over post
+  // view iteration domains when resolving the parallel and loop map concrete
+  // IDs.
+  std::unordered_set<IterDomain*> rfactor_concrete_count_reset_domains;
+
+  // Gather all dimensions that should be considered having one concrete
+  // dimension, this is propagated from the domains in
+  // rfactor_concrete_count_reset_domains as we're building the map. This will
+  // reset all domains topologically before the domains in
+  // rfactor_concrete_count_reset_domains to have count of 1 concrete domain.
+  // Preventing pre-view ID's from being involved in concrete resolution when
+  // mapped to post view domains in the current map (Parallel and Loop maps
+  // IndexMap is valid with any domain in any disjoint set being the concrete
+  // ID).
+  std::unordered_set<IterDomain*> count_one_concrete_dims;
+
+  // Should be run on consumer id after producer id and consumer id are mapped
+  // together, will place all id's in consumer_id's disjoint set into
+  // count_one_concrete_dims if consumer_id is in
+  // rfactor_concrete_count_reset_domains
+  auto maybePropagateConcreteCountOne = [&](IterDomain* consumer_id) {
```

Comment:
I'd prefer to have this as a separate function rather than a lambda. Same for `pullConcreteCountResetIds` too.

Reply:
Hmm, noticed `pullConcreteCountResetIds` is also changed to take everything using `[&]` in the next PR. I'd be really hesitant to do so except for trivial lambdas.

Reply:
Sidenote: Is the naming convention for lambdas like a regular function or a variable? 
e.g., `maybePropagateConcreteCountOne` or `maybe_propagate_concrete_count_one`

Reply:
My general comment is to avoid a lambda if you need to name it with multiple words.

Reply:
Will pull lambdas, was doing it just for convenience in development/testing.